src/extractors/htmlcharacterextractor.py

```diff
@@ -0,0 +1,36 @@
+import json
+from pathlib import Path
+import requests
+from typing import List
+
+from .characterfactory import Character
+from .extractor import Extractor
+
+class HtmlCharacterExtractor(Extractor):
+    __characters: List[Character]
+
+    def __init__(self):
+        self.__characters = []
+
+    def __fetch_data(self) -> None:
+        response: requests.Response = requests.get(
+            "https://html.spec.whatwg.org/entities.json", timeout=120
+        )
+        entities = json.loads(response.text)
+        for key in entities:
+            if key.endswith(';') and key != "&NewLine;":
+                name = key[1:-1]
+                char = ''.join(list(map(chr, entities[key]['codepoints'])))
+                self.__characters.append(Character(char, name))
```

Comment:
I wouldn't use the `Character` data class here, as that is meant for exactly one character (this also explains why it can't cope with a string of more than one character 😉)

You could define your own data class, similar to what `KaomojiExtractor` does.

Reply:
Going by https://unicode.org/glossary/ ("Character (...) (2) Synonym for abstract character." and https://www.unicode.org/versions/Unicode15.1.0/ch03.pdf#G2212 ("Abstract characters not directly encoded by the Unicode Standard can often be represented by the use of combining character sequences") it didn't seem outlandish to me, but indeed the class might not be adding much in this case, will update.